R/pool.R

```diff
@@ -45,118 +53,178 @@ Pool <- R6Class("Pool",
       }
       reg.finalizer(self,
         function(self) {
-          protectDefaultScheduler({
-            freeEnv <- private$freeObjects
-            if (length(freeEnv) > 0) {
-              self$close()
-            }
-          })
+          if (self$valid) {
+            warning("Closing leaked pool.")
+            self$close()
+          }
         },
         onexit = TRUE)
```

Comment:
Hmmm, maybe our finalizers should have `onexit = FALSE`. I can't decide which is worse: trying to exit an R process and waiting on DB connections to close, or exiting without cleaning up some kind of external state. Thoughts?